src/main/c/Objects/pyjobject.c

```diff
@@ -516,49 +525,31 @@ static PyMethodDef pyjobject_methods[] = {
 
 static PyMemberDef pyjobject_members[] = {
     {"__dict__", T_OBJECT, offsetof(PyJObject, attr), READONLY},
+    {"__dictoffset__", T_PYSSIZET, offsetof(PyJObject, attr), READONLY}, // NOTE: Sets `PyTypeObject.tp_dictoffset` for heap types...
```

Comment:
This is an interesting replacement for `tp_dictoffset`......
Looks like this is actually used in the internals somewhere...

Oddly, the heap-type API is mostly undocumented....